locust/test/test_main.py

```diff
@@ -3,30 +3,218 @@
+
+    if platform.system() == "Windows":
+        while True:
+            stdout_line = proc.stdout.readline()
+            if not stdout_line:
+                break
+            output.append(stdout_line)
+            if "All users spawned" in stdout_line:
+                return True
+            stderr_line = proc.stderr.readline()
+            if stderr_line:
+                output.append(stderr_line)
+    else:
+        ready_to_read, _, _ = select.select([proc.stdout, proc.stderr], [], [], 0.1)
+        for pipe in ready_to_read:
+            pipe_line = pipe.readline()
+            output.append(pipe_line)
+            if "All users spawned" in pipe_line:
+                return True
+    return False
+
+
+def read_output_non_blocking(proc, output_list):
+    """
+    Reads real-time output from a subprocess in a non-blocking way using gevent.
+    :param proc: Subprocess object
+    :param output_list: List to store the real-time output.
+    """
+    while proc.poll() is None:  # Keep reading while the process is still running
+        line = proc.stdout.readline()
+        if line:
+            output_list.append(line)
+            print(line.strip())  # Optional: for real-time feedback
+        gevent.sleep(0.1)  # Yield control to allow other greenlets to run
+
+
+def wait_for_output_condition_non_threading(proc, output_lines, condition, timeout=30):
+    start_time = time.time()
+    while True:
+        combined_output = "\n".join(output_lines)
+        if condition in combined_output:
+            return True
+        if time.time() - start_time >= timeout:
+            return False
+        gevent.sleep(0.1)
+
+
+class ProcessManager:
+    def __init__(self, proc, stdout_reader, stderr_reader, temp_file_path, output_lines):
+        self.proc = proc
+        self.stdout_reader = stdout_reader
+        self.stderr_reader = stderr_reader
+        self.temp_file_path = temp_file_path
+        self.output_lines = output_lines
+
+    def cleanup(self):
+        if self.temp_file_path is not None and os.path.exists(self.temp_file_path):
+            os.unlink(self.temp_file_path)
+
+
+@contextmanager
+def run_locust_process(file_content=None, args=None, port=None):
+    temp_file_path = None
+    locust_command = ["locust"]
+
+    if file_content is not None:
+        with NamedTemporaryFile(mode="w+", delete=False, suffix=".py") as temp_file:
+            temp_file.write(file_content)
+            temp_file.flush()
+            temp_file_path = temp_file.name
+        locust_command += ["-f", temp_file_path]
+
+    if args:
+        locust_command += args
+
+    if port is not None:
+        locust_command += ["--web-port", str(port)]
+
+    proc = subprocess.Popen(
+        locust_command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        bufsize=1,
+        text=True,
+        env={**os.environ, "PYTHONUNBUFFERED": "1"},
```

Comment:
This is probably unnecessary, because you're setting it on this line too https://github.com/locustio/locust/pull/2899/files#diff-81a8e34f7c9a9b06d20ff25a8b8d6f6fe983707f8ba66780659df4fe4fffa2aaR247